app/src/main/java/com/google/samples/apps/sunflower/adapters/GardenPlantingAdapter.kt

```diff
@@ -21,19 +21,23 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.databinding.DataBindingUtil
-import androidx.navigation.findNavController
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
-import com.google.samples.apps.sunflower.GardenFragmentDirections
 import com.google.samples.apps.sunflower.R
 import com.google.samples.apps.sunflower.data.PlantAndGardenPlantings
 import com.google.samples.apps.sunflower.databinding.ListItemGardenPlantingBinding
+import com.google.samples.apps.sunflower.event.Event
 import com.google.samples.apps.sunflower.viewmodels.PlantAndGardenPlantingsViewModel
 
 class GardenPlantingAdapter(
     val context: Context
 ) : ListAdapter<PlantAndGardenPlantings, GardenPlantingAdapter.ViewHolder>(GardenPlantDiffCallback()) {
 
+    private val _openPlant = MutableLiveData<Event<String>>()
+    internal val openPlant: LiveData<Event<String>> = _openPlant
```

Comment:
I don't see why this has to be a LiveData vs a lambda or a Listener. I get that the Event type allows for one-time handled events, so this would operate similar to a lambda callback or Listener, and that there's an added benefit that a fragment/activity would only execute the attached logic if it was in an appropriate lifecycle state. But, is a click happening during a bad lifecycle state a real possibility here, can it not be handled with a lifecycle check by the owner, and is it too much complexity vs just setting a listener / using a lambda in the constructor?

Reply:
@mcastagnolo how about #290 ?